### DISCUSSION TITLE
Import SVG that does not contain 'content' attribute in <svg> element

### DISCUSSION THREAD

**Discussion by pawelciupkaOnwelo** [jgraph/drawio#4042] [Q&A] [answered]
I have a file in SVG format that starts the same as the SVG exported from diagrams.net. 
`<?xml version="1.0" encoding="UTF-8"? >`
`<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd" >`
`<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1" width="100px" height="100px" >`
The only difference to the SVG files generated by diagrams.net is the **content** attribute, which does not exist in my file.

While I am opening this file I am getting this error:
![image](https://github.com/jgraph/drawio/assets/120187777/b1ec8ed0-f28c-4181-9850-a57cb88b195f)

My question is - is it possible to somehow generate **content** attribute based on existing SVG file? Or it is another possibly to import SVG file to diagrams.net and starts editing.

Thanks!

**Comment by davidjgraph** [comment]
How are you importing your SVG file? Can you attach it?

**Reply by pawelciupkaOnwelo** [reply]
I am using "Open Existing Diagram" option when I am opening diagrams.net website:
![image](https://github.com/jgraph/drawio/assets/120187777/b47a933a-af73-4bcb-9fc9-ac985b8c2ee4)

Here is the files:
![test_svg](https://github.com/jgraph/drawio/assets/120187777/68253a58-b632-4f19-a003-fa504e824239)

**Reply by davidjgraph** [reply]
If there no data in the SVG it isn't a draw.io file, so the message is correct. SVG can only be imported.

**Reply by pawelciupkaOnwelo** [reply]
What you mean by 'no data in the SVG'? Like I mentioned previosuly, in my example there is no **content** attribute in the <svg> element from line 3.
I am not sure if file is visible correctly, so at the bottom I am uploading a screenshot of the contents of the above SVG.
![image](https://github.com/jgraph/drawio/assets/120187777/58cc2d30-8cc8-4061-813e-ad0daf467bcd)

**Reply by davidjgraph** [reply]
No diagram data, i.e. it's not a native draw.io file.

**Reply by pawelciupkaOnwelo** [reply]
Is it possible to add diagram data based on this svg, to be able to open this in diagrams.net?

**Reply by davidjgraph** [reply]
Save the diagram with the XML embedded in the SVG.

**Reply by pawelciupkaOnwelo** [reply]
But how can I transform svg to XML that will be acceptable in **content** attribute. I can see that there is using mxgraph. I tried to use https://github.com/jgraph/svg2xml/tree/merge but it is not working. 

Also I am not able to load my svg to diagrams.net and then export as a xml.

**Reply by davidjgraph** [reply, marked as answer]
You'd have to write this yourself, it will be huge task.

> Also I am not able to load my svg to diagrams.net and then export as a xml.

No, draw.io is not an SVG editor.

**Reply by pawelciupkaOnwelo** [reply]
Thank you for your answers.
